cmd/query/app/static_handler.go

```diff
@@ -94,11 +126,54 @@ func NewStaticAssetsHandler(staticAssetsRoot string, options StaticAssetsHandler
 		}
 		indexBytes = basePathPattern.ReplaceAll(indexBytes, []byte(fmt.Sprintf(basePathReplace, options.BasePath)))
 	}
-	return &StaticAssetsHandler{
-		options:   options,
-		indexHTML: indexBytes,
-		assetsFS:  assetsFS,
-	}, nil
+
+	return indexBytes, nil
+}
+
+func (sH *StaticAssetsHandler) watch() {
+	if sH.options.UIConfigPath == "" {
+		return
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		sH.options.Logger.Error("failed to create a new watcher for the UI config", zap.Error(err))
+		return
+	}
+	defer watcher.Close()
```

Comment:
I'd recommend the  above to be done in the constructor, so that go-routine is not even started until watcher is created  successfully

Reply:
I kept it here, and changed the constructor to not execute `watch()` as a go-routine. Should have the same effect, but the watcher is now encapsulated where it's needed.